python/lsst/daf/butler/core/dataUnit.py

```diff
@@ -275,6 +331,31 @@ def _initDataUnits(self, config, builder):
                                 link=link)
             self[dataUnitName] = dataUnit
 
+    def _initDataUnitJoins(self, config, builder):
+        """Initialize `DataUnit` join entries.
+
+        Parameters
+        ----------
+        config : `SchemaConfig`
+            Schema configuration describing `DataUnit` join relations.
+        builder : `SchemaBuilder`, optional
+            When given, create `sqlalchemy.core.Table` entries for every `DataUnit` table.
+        """
+        for dataUnitJoinName, dataUnitJoinDescription in config.items():
+            table = None
+            if 'tables' in dataUnitJoinDescription and builder is not None:
+                for tableName, tableDescription in dataUnitJoinDescription['tables'].items():
+                    table = builder.addTable(tableName, tableDescription)
+            lhs = dataUnitJoinDescription.get('lhs', None)
+            rhs = dataUnitJoinDescription.get('rhs', None)
+            summarizes = dataUnitJoinDescription.get('summarizes', None)
+            dataUnitJoin = DataUnitJoin(name=dataUnitJoinName,
+                                        lhs=lhs,
+                                        rhs=rhs,
+                                        summarizes=summarizes,
+                                        table=table)
+            self.joins[dataUnitJoinName] = dataUnitJoin
```

Comment:
I don't think this needs to be changed here, but as an FYI, on DM-14428 I'll probably switch the keys to be `(lhs, rhs)`,  as I think that's what we'll want to use for lookup most of the time.  Please shout if you think that'll be problematic.

Reply:
I think that is fine.